src/components/MediaPlayer/VideoJSPlayer.js

```diff
@@ -116,66 +141,197 @@ function VideoJSPlayer({
+        manifestDispatch({ item: firstItem, type: 'switchItem' });
+      }
     }
-  }, [startTime, endTime, isClicked, isReady]);
+  }, [currentNavItem, isReady]);
 
+  /**
+   * Setting the current time of the player when using structure navigation
+   */
   React.useEffect(() => {
-    if (isClicked && canvasIndex !== cIndex) {
-      switchPlayer();
+    if (player !== null && isReady) {
+      player.currentTime(currentTime, playerDispatch({ type: 'resetClick' }));
     }
-    setCIndex(canvasIndex);
-  }, [canvasIndex]);
+  }, [isClicked]);
+
+  /**
+   * Remove existing timerail highlight if the player's currentTime
+   * doesn't fall within a defined structure item
+   */
+  React.useEffect(() => {
+    if (!player || !currentPlayer) {
+      return;
+    } else if (isContained == false && player.markers) {
+      player.markers.removeAll();
+    }
+  }, [isContained]);
+
+  /**
+   * Handle the 'seeked' event when player's scrubber or progress bar is
+   * used to change the currentTime.
+   */
+  const handleSeeked = () => {
+    if (player !== null && isReadyRef.current) {
+      const seekedTime = player.currentTime();
+      playerDispatch({
+        currentTime: seekedTime,
+        type: 'setCurrentTime',
+      });
+      // Find the relevant media segment from the structure
+      const isInStructure = getActiveSegment(seekedTime);
+
+      if (isInStructure) {
+        setIsContained(true);
+        manifestDispatch({ item: isInStructure, type: 'switchItem' });
+      } else {
+        setIsContained(false);
+      }
+    }
+  };
 
+  /**
+   * Handle the 'ended' event fired by the player when a section comes to
+   * an end. If there are sections ahead move onto the next canvas and
+   * change the player and the state accordingly.
+   */
   const handleEnded = () => {
     if (hasNextSection({ canvasIndex, manifest })) {
       manifestDispatch({ canvasIndex: canvasIndex + 1, type: 'switchCanvas' });
-      // Reset startTime to zero
+
+      // Reset startTime and currentTime to zero
       playerDispatch({ startTime: 0, type: 'setTimeFragment' });
+      playerDispatch({ currentTime: 0, type: 'setCurrentTime' });
+
       // Update the current nav item to next item
-      manifestDispatch({
-        item: getNextItem({ canvasIndex, manifest }),
-        type: 'switchItem',
-      });
+      const nextItem = getNextItem({ canvasIndex, manifest });
+
+      const { start } = getMediaFragment(getItemId(nextItem));
+
+      // If there's a structure item at the start of the next canvas
+      // mark it as the currentNavItem. Otherwise empty out the currentNavItem.
+      if (start == 0) {
+        setIsContained(true);
+        manifestDispatch({
+          item: nextItem,
+          type: 'switchItem',
+        });
+      } else {
+        manifestDispatch({ item: null, type: 'switchItem' });
+      }
 
       handleIsEnded();
 
       setCIndex(cIndex + 1);
     }
   };
 
+  /**
```

Comment:
These comments are great.  Nice work

Reply:
Yeah, I thought it would be helpful to anyone contributing to the code in the future (including me :D). We didn't have much time to add comments when we were working on the community sprint, so I had to spent some time on understanding what the code is doing in a couple places.